packages/connect-node/src/connect-transport.ts

```diff
@@ -206,10 +206,11 @@ export function createConnectTransport(
               opt.readMaxBytes
             );
           }
-          if (isConnectUnaryError) {
+          if (isUnaryError) {
             throw errorFromJsonBytes(
               responseBody,
-              appendHeaders(header, trailer)
+              appendHeaders(header, trailer),
+              unaryError
```

Comment:
We ran into https://github.com/bufbuild/connect-es/issues/469 with fastify, which replies with an application/json content type for a HTTP 404 when application/json was sent.

The logic was changed to raise a ConnectError for the HTTP status code when parsing fails to fix the issue. This change is applied to the transports from @bufbuild/connect-web and @bufbuild/connect-node.